client/src/components/CategoriesAccordion.jsx

```diff
@@ -1,15 +1,29 @@
-import { useRef, useState } from "react";
+import { useEffect, useRef, useState } from "react";
 
 const CategoriesAccordion = ({
   categories,
   item,
   subCategories,
   setSubCategories,
+  id,
 }) => {
-  const [isOpened, setOpened] = useState(false);
-
+  const [isOpened, setOpened] = useState(
+    parseInt(id) === item.id ? true : false
+  );
   const contentElement = useRef(null);
 
+  useEffect(() => {
+    const idList = categories
+      .map((i) => {
+        if (i.supercategoryId === parseInt(id)) {
+          return i.id.toString();
+        }
+      })
+      .filter((i) => i !== undefined);
```

Comment:
to avoid this if, this can be
```suggestion
      const idList = categories
      .filter(cat => cat && cat.supercategoryId === parseInt(id))
      .map(cat => cat.toString())
      );
```